bin/compile

```diff
@@ -92,149 +98,168 @@ if [[ ! -f Procfile ]]; then
+# shellcheck source=bin/steps/eggpath-fix
+source "$BIN_DIR/steps/eggpath-fix"
 
 # Mercurial support.
-source $BIN_DIR/steps/mercurial
+# shellcheck source=bin/steps/mercurial
+source "$BIN_DIR/steps/mercurial"
 
 # Pylibmc support.
-source $BIN_DIR/steps/pylibmc
+# shellcheck source=bin/steps/pylibmc
+source "$BIN_DIR/steps/pylibmc"
 
 # Libffi support.
-source $BIN_DIR/steps/cryptography
+# shellcheck source=bin/steps/cryptography
+source "$BIN_DIR/steps/cryptography"
 
 # Support for Geo libraries.
-sub-env $BIN_DIR/steps/geo-libs
+# shellcheck source=bin/steps/geo-libs
+sub-env "$BIN_DIR/steps/geo-libs"
 
 # GDAL support.
-source $BIN_DIR/steps/gdal
+# shellcheck source=bin/steps/gdal
+source "$BIN_DIR/steps/gdal"
 
 # Uninstall removed dependencies with Pip.
 let start=$(nowms)
-source $BIN_DIR/steps/pip-uninstall
+# shellcheck source=bin/steps/pip-uninstall
+source "$BIN_DIR/steps/pip-uninstall"
 mtime "pip.uninstall.time" "${start}"
 
 # Install dependencies with Pip (where the magic happens).
 let start=$(nowms)
-source $BIN_DIR/steps/pip-install
+# shellcheck source=bin/steps/pip-install
+source "$BIN_DIR/steps/pip-install"
 mtime "pip.install.time" "${start}"
 
 # Support for NLTK corpora.
 let start=$(nowms)
-sub-env $BIN_DIR/steps/nltk
+sub-env "$BIN_DIR/steps/nltk"
 mtime "nltk.download.time" "${start}"
 
 # Support for pip install -e.
 # In CI, $BUILD_DIR is /app.
 if [[ ! "$BUILD_DIR" == "/app" ]]; then
-  rm -fr $BUILD_DIR/.heroku/src
-  deep-cp /app/.heroku/src $BUILD_DIR/.heroku/src
+  rm -fr "$BUILD_DIR/.heroku/src"
+  deep-cp /app/.heroku/src "$BUILD_DIR/.heroku/src"
 fi
 
 
 # Django collectstatic support.
 let start=$(nowms)
-sub-env $BIN_DIR/steps/collectstatic
+sub-env "$BIN_DIR/steps/collectstatic"
 mtime "collectstatic.time" "${start}"
 
 # Create .profile script for application runtime environment variables.
-set-env PATH '$HOME/.heroku/python/bin:$PATH'
+set-env PATH "\$HOME/.heroku/python/bin:\$PATH"
 set-env PYTHONUNBUFFERED true
 set-env PYTHONHOME /app/.heroku/python
-set-env LIBRARY_PATH '/app/.heroku/vendor/lib:/app/.heroku/python/lib:$LIBRARY_PATH'
-set-env LD_LIBRARY_PATH '/app/.heroku/vendor/lib:/app/.heroku/python/lib:$LD_LIBRARY_PATH'
+
+set-env LIBRARY_PATH "/app/.heroku/vendor/lib:/app/.heroku/python/lib:\$LIBRARY_PATH"
+set-env LD_LIBRARY_PATH "/app/.heroku/vendor/lib:/app/.heroku/python/lib:\$LD_LIBRARY_PATH"
+
 set-default-env LANG en_US.UTF-8
 set-default-env PYTHONHASHSEED random
 set-default-env PYTHONPATH /app/
 
 # Install sane-default script for $WEB_CONCURRENCY and $FORWARDED_ALLOW_IPS.
-cp $ROOT_DIR/vendor/WEB_CONCURRENCY.sh $WEB_CONCURRENCY_PROFILE_PATH
-cp $ROOT_DIR/vendor/python.gunicorn.sh $GUNICORN_PROFILE_PATH
+cp "$ROOT_DIR/vendor/WEB_CONCURRENCY.sh" "$WEB_CONCURRENCY_PROFILE_PATH"
+cp "$ROOT_DIR/vendor/python.gunicorn.sh" "$GUNICORN_PROFILE_PATH"
 
 
 # Experimental post_compile hook.
-source $BIN_DIR/steps/hooks/post_compile
+# shellcheck source=bin/steps/hooks/post_compile
```

Comment:
they tell shellcheck how to check sources that come from variables. standard practice.